Lib/pathlib/__init__.py

```diff
@@ -1105,7 +1105,7 @@ def copy(self, target, **kwargs):
         if not hasattr(target, 'with_segments'):
             target = self.with_segments(target)
         ensure_distinct_paths(self, target)
-        target._copy_from(self, **kwargs)
+        list(target._copy_from(self, **kwargs))  # Consume generator.
```

Comment:
per the docs in itertools the best way to consume a generator is collections.deque:

```
def consume(iterator, n=None):
    "Advance the iterator n-steps ahead. If n is None, consume entirely."
    # Use functions that consume iterators at C speed.
    if n is None:
        deque(iterator, maxlen=0)
    else:
        next(islice(iterator, n, n), None)
```
```suggestion
        collections.deque(target._copy_from(self, **kwargs), maxlen=0)  # Consume generator.
```